src/plugin/plugin.ts

```diff
@@ -32,6 +42,62 @@ export const createPlugin = <
 ): Plugin<PluginState, Schema> => {
   return new Plugin<PluginState, Schema>({
     key: pluginKey,
+    /**
+     * Update the elements to represent the current selection.
+     */
+    appendTransaction: (_, oldState, newState) => {
+      // If we are not transitioning between at least one selection of non-zero
+      // width, we cannot be altering the element selection state.
+      if (selectionIsZeroWidth(oldState) && selectionIsZeroWidth(newState)) {
+        return;
+      }
+
+      const tr = newState.tr;
+      const elementNodeToPos = new Map<Node, number>();
+
+      // Find all the nodes within the current selection.
+      newState.doc.nodesBetween(
+        newState.selection.from,
+        newState.selection.to,
+        (node, pos) => {
+          if (
+            isProseMirrorElement(node) &&
+            newState.selection.from <= pos &&
+            newState.selection.to >= pos + node.nodeSize
+          ) {
+            elementNodeToPos.set(node, pos);
+            return false;
+          }
+        }
+      );
+
+      // Update nodes representing any elements that are no longer selected.
+      newState.doc.descendants((node, pos) => {
+        if (isProseMirrorElementSelected(node) && !elementNodeToPos.get(node)) {
+          if (node.attrs[elementSelectedNodeAttr] === true) {
```

Comment:
There is some redundancy here as `isProseMirrorElementSelected(node)` and `node.attrs[elementSelectedNodeAttr] === true` are equivalent.